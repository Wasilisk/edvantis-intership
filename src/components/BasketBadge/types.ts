export type BasketBadgeProps = {
    count: number
}