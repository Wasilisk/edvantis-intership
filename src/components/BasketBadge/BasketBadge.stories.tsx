import {Story} from "@storybook/react";
import React from "react";
import BasketBadge from "./index";
import {BasketBadgeProps} from "./types";

export default {
    title: 'Application/BasketBadge',
    component: BasketBadge,
};

const Template: Story<BasketBadgeProps> = (props) => <BasketBadge count={props.count}/>;

export const Default = Template.bind({});
Default.args = {
    count: 2
};