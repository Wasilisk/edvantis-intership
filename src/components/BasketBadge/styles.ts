import styled from "styled-components";

export const BasketBadgeStyles = styled.div`
  width: 70px;
  height: 70px;
  border-radius: 50%;
  box-shadow: 0 0 1px rgb(33 39 54 / 4%), 0 2px 4px rgb(33 39 54 / 16%);
  position: fixed;
  top: 80px;
  right: 30px;
  background-color: #ffffff;
  padding: 15px;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  z-index: 1;
  
  svg {
    height: 100%;
    width: 100%;
  }
  
  .counter {
    background-color: #7bb57c;
    color: #ffffff;
    top: -2px;
    right: -2px;
    display: flex;
    align-items: center;
    justify-content: center;
    position: absolute;
    width: 24px;
    height: 24px;
    border-radius: 50%;
  }
`;