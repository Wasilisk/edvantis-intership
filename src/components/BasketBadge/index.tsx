import React, {useRef, useState} from 'react';
import {BasketBadgeStyles} from './styles';
import {ReactComponent as BasketIcon} from "../../assets/icons/basket.svg"
import {BasketBadgeProps} from "./types";
import MainPopup from "../Popups/MainPopup";
import {useDisableBodyScroll} from "../../hooks";
import ShoppingCart from "../ShoppingCart";
import {CSSTransition} from "react-transition-group";

const BasketBadge = ({count}: BasketBadgeProps) => {
    const [isPopupOpen, setIsPopupOpen] = useState<boolean>(false);
    const nodeRef = useRef<HTMLDivElement>(null);

    const openModal = () => setIsPopupOpen(true);
    const closeModal = () => setIsPopupOpen(false);

    useDisableBodyScroll(isPopupOpen);

    return (
        <>
            <CSSTransition
                in={Boolean(count)}
                timeout={500}
                classNames="basket"
                unmountOnExit
                nodeRef={nodeRef}
            >
            <BasketBadgeStyles>
                <BasketIcon onClick={openModal} data-testid="basket-icon"/>
                <div className="counter" data-testid="counter">
                    {count}
                </div>
            </BasketBadgeStyles>
            </CSSTransition>
            <MainPopup title="Shopping cart" isOpen={isPopupOpen} onClose={closeModal}>
                <ShoppingCart closeModal={closeModal}/>
            </MainPopup>
        </>
    );
};

export default BasketBadge;