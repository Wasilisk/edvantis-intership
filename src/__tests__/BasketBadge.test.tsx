import {render} from "./__utils__/store-wrapper";
import BasketBadge from "../components/BasketBadge";

describe('BasketBadge component', () => {
    it('should render header component', () => {
        const {asFragment, getByTestId} = render(<BasketBadge count={1}/>);

        const basketIcon = getByTestId('basket-icon');
        expect(basketIcon).toBeDefined();

        const counter = getByTestId('counter');
        expect(Number(counter.textContent)).toBe(1);

        expect(asFragment()).toMatchSnapshot();
    })
    it('should no display if counter is zero', () => {
        const {queryByTestId} = render(<BasketBadge count={0}/>);

        expect(queryByTestId('basket-icon')).toBeFalsy();

        expect(queryByTestId('counter')).toBeFalsy();
    })
})