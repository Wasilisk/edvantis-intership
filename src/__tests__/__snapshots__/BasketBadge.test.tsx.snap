// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketBadge component should render header component 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ gPmtV"
  >
    <svg
      data-testid="basket-icon"
    >
      basket.svg
    </svg>
    <div
      class="counter"
      data-testid="counter"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;
